### PR TITLE
Include GFA assembly graphs in results

### LINKED ISSUE #46: Are GFA outputs from de novo assembly useful?
Created this issue mostly to address this TODO item
[`// TODO nf-core: Rename and output gfa files too?`](https://github.com/nf-core/viralrecon/blob/dev/main.nf#L1381)

### PR DESCRIPTION
**Pull request by heuermh**
Fixes #46 

I also wonder, for consistency's sake, if the `${sample}.scaffolds.fa` and `${sample}.meta.scaffolds.fa` files in results for spades and metaspades respectively should be renamed to match `${sample}.assembly.fa` from unicycler?

## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If necessary, also make a PR on the [nf-core/viralrecon branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/viralrecon)
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Documentation in `docs` is updated
- [ ] `CHANGELOG.md` is updated
- [ ] `README.md` is updated

**Learn more about contributing:** [CONTRIBUTING.md](https://github.com/nf-core/viralrecon/tree/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -1474,6 +1474,7 @@ process SPADES {
                                                                  ch_spades_blast,
                                                                  ch_spades_abacas,
                                                                  ch_spades_plasmidid
+    set val(sample), val(single_end), file("*scaffolds.fa"), file("*assembly.gfa") into ch_spades_graph
```
Comment (drpatelh) [inline main.nf:1477]:
```suggestion
    file "*assembly.gfa" into ch_spades_graph

```

File: main.nf [hunk 2]
```diff
@@ -1787,6 +1791,7 @@ process UNICYCLER {
                                                                 ch_unicycler_blast,
                                                                 ch_unicycler_abacas,
                                                                 ch_unicycler_plasmidid
+    set val(sample), val(single_end), file("*assembly.fa"), file("*assembly.gfa") into ch_unicycler_graph
```
Comment (drpatelh) [inline main.nf:1794]:
```suggestion
    file "*assembly.gfa" into ch_unicycler_graph

```

File: main.nf [hunk 3]
```diff
@@ -1630,6 +1632,7 @@ process METASPADES {
                                                                  ch_metaspades_blast,
                                                                  ch_metaspades_abacas,
                                                                  ch_metaspades_plasmidid
+    set val(sample), val(single_end), file("*scaffolds.fa"), file("*assembly.gfa") into ch_metaspades_graph
```
Comment (drpatelh) [inline main.nf:1635]:
```suggestion
    file "*assembly.gfa" into ch_metaspades_graph

```

File: main.nf (hunk 1 above)
Comment (drpatelh) [reply main.nf:1477]: Doesnt look like these files are being used in any downstream processes yet so better to not give them a named channel 👍

**Comment by drpatelh** [comment]
Thanks @heuermh :) Please see minor comments above.

**Comment by drpatelh** [comment]
Fast-tracked this so I can add in the `minimap2`, `seqwish` and `vg` steps for the other assemblies ;)

**Comment by heuermh** [comment]
Thank you, @drpatelh!

**Pull request merged**